### PR TITLE
[MM-21789] Update role on initial add of group to a team or channel

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Passing the role on initial add of a group to a channel or team configuration page was forgotten. 
This PR adds it so the role is saved along with linking the group to that channel or team.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21789

### CODE REVIEW

File: components/admin_console/team_channel_settings/team/details/team_details.jsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ export default class TeamDetails extends React.Component {
                 filter((g) => {
                     return !origGroups.some((group) => group.id === g.id);
                 }).
-                map((g) => actions.linkGroupSyncable(g.id, teamID, Groups.SYNCABLE_TYPE_TEAM, {auto_add: true}));
+                map((g) => actions.linkGroupSyncable(g.id, teamID, Groups.SYNCABLE_TYPE_TEAM, {auto_add: true, scheme_admin: g.scheme_admin}));
```
Comment (ethervoid) [inline components/admin_console/team_channel_settings/team/details/team_details.jsx]: We should add some tests for the link/unlink/patch logic

Comment (hahmadia) [reply components/admin_console/team_channel_settings/team/details/team_details.jsx]: Done! Thanks

File: e2e/cypress/integration/enterprise/system_console/team_and_channel_asign_roles_spec.js [hunk 2]
```diff
@@ -81,6 +81,82 @@ describe('System Console', () => {
...
+        // * Ensure default role is Member
+        cy.findByTestId('current-role').should('have.text', 'Member').click();
+
+        // * Assert that only one option exists in the dropdown for changing roles
+        cy.get('#role-to-be-menu').then((el) => {
+            expect(el[0].firstElementChild.children.length).equal(1);
+        });
+
+        // # Continue changing the role to Team Admin
+        cy.get('#role-to-be').click();
```
Comment (srkgupta) [inline e2e/cypress/integration/enterprise/system_console/team_and_channel_asign_roles_spec.js]: Hi @hahmadia 
At times, when the Cypress test clicks on the dropdown to change the role, the Save button does not gets enabled, even if the role dropdown was selected. It looks like the trigger action when the role is changed is not properly triggered when we click through Cypress test. Can you please check and update it throughout the test.

**Comment by srkgupta** [comment]
@hahmadia  Can you please resolve the merge conflict.

**Pull request merged**